client/lib/domains/resolve-domain-status.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import moment from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import { transferStatus, type as domainTypes } from './constants';
+import { isExpiringSoon } from 'lib/domains/utils/is-expiring-soon';
+import { isRecentlyRegistered } from 'lib/domains/utils/is-recently-registered';
+import { hasPendingGSuiteUsers } from 'lib/gsuite';
+import { shouldRenderExpiringCreditCard } from 'lib/purchases';
+
+export function resolveDomainStatus(
```

Comment:
This is nice 👍. I like how you have extracted all the similarly named functions that "resolve status" for the different domain types and integrated it into one function that works for all types. It is no doubt neater.

However in this PR, or a subsequent one, I think we should further extract the cases into different functions in this file to make the case statements easier to read and decipher at a glance.